src/ReaderScreen/index.jsx

```diff
@@ -143,17 +221,39 @@ const Reader = ({ navigation, route }) => {
   const handleLoadStart = useCallback(() => {
     setTimeout(() => {
       toggleViewLoaded(true);
-    }, 500);
+    }, 100);
   }, []);
 
   const handleError = useCallback((syntheticEvent) => {
     const { nativeEvent } = syntheticEvent;
     logError(`Reader web View Error ${nativeEvent}`);
+    setNeedsReload(true);
   }, []);
 
   const handleHttpError = useCallback((syntheticEvent) => {
     const { nativeEvent } = syntheticEvent;
     logError("HTTP error status code:", nativeEvent.statusCode);
+    setNeedsReload(true);
+  }, []);
+
+  const handleLoadEnd = useCallback(() => {
+    lastActivityTimestamp.current = Date.now();
+    setNeedsReload(false);
+
+    // Set initial position after load
+    if (currentPosition > 0 && webViewRef.current) {
+      webViewRef.current.postMessage(
+        JSON.stringify({
+          setPosition: currentPosition,
```

Comment:
I couldn't find a handler for setPosition in the webview script. Can you please point me to where we are using this property?